mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBEntityMappingValidator.java

```diff
@@ -33,8 +48,76 @@ public void validateMapping(SchemaDefinitionContext context) {
 		validateEntityCollectionNames( context.getAllEntityKeyMetadata() );
 		validateAssociationNames( context.getAllAssociationKeyMetadata() );
 		validateAllPersisters( context.getSessionFactory().getEntityPersisters().values() );
+		validateIndexSpecs( context );
+	}
+
+	@Override
+	public void initializeSchema( SchemaDefinitionContext context) {
+		for(IndexSpec indexSpec : indexSpecs)
+		{
+
+			OgmDialect ogmDialect = (OgmDialect) context.getSessionFactory()
+					.getDialect();
+			ogmDialect.getGridDialect().createIndex(indexSpec);
+		}
+		String toto = "toto";
+	}
+
+	private void validateIndexSpecs( SchemaDefinitionContext context ) {
+		indexSpecs= new ArrayList<>();
+		Database database = context.getDatabase();
+		for(Table table : database.getDefaultNamespace().getTables())
+		{
+			Iterator<UniqueKey> keys = table.getUniqueKeyIterator();
+			while(keys.hasNext())
+			{
+				IndexSpec indexSpec = new IndexSpec( keys.next() );
+				validateIndexSpec(indexSpec);
+				indexSpecs.add(indexSpec);
```

Comment:
I wonder if we should add a `preProcessing(...)` method to initialize some variables. I don't like too much that the validation method is also responsible for the initialization of the fields